lightning/src/ln/channelmanager.rs

```diff
@@ -2878,15 +2880,22 @@ impl<'a> PersistenceNotifierGuard<'a, fn() -> NotifyOption> {
 		PersistenceNotifierGuard {
 			event_persist_notifier: &cm.get_cm().event_persist_notifier,
 			needs_persist_flag: &cm.get_cm().needs_persist_flag,
-			should_persist: persist_check,
+			should_persist: Some(persist_check),
 			_read_guard: read_guard,
 		}
 	}
 }
 
-impl<'a, F: FnMut() -> NotifyOption> Drop for PersistenceNotifierGuard<'a, F> {
+impl<'a, F: FnOnce() -> NotifyOption> Drop for PersistenceNotifierGuard<'a, F> {
 	fn drop(&mut self) {
-		match (self.should_persist)() {
+		let should_persist = match self.should_persist.take() {
+			Some(should_persist) => should_persist,
+			None => {
+				debug_assert!(false);
+				return;
```

Comment:
Silent return in release mode - if this would really happen, do you really want to keep running without any indication in logs etc?